smart-contract-verifier/src/compiler/version.rs

```diff
@@ -10,50 +11,58 @@ pub enum ParseError {
     Parse(String),
     #[error("couldn't parse commit hash")]
     CommitHash(hex::FromHexError),
+    #[error("cannot parse semver: {0}")]
+    Semver(#[from] semver::Error),
 }
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub struct ReleaseVersion {
     pub version: semver::Version,
-    pub commit: [u8; 4],
+    pub commit: String,
 }
 
 impl FromStr for ReleaseVersion {
     type Err = ParseError;
 
     /// Parses release version from string formated as
-    /// `(v)*VERSION*+commit.*COMMITHASH*`, examples:
+    /// `(v)*VERSION*(-*PRERELEASE*)+commit.*COMMITHASH*`, examples:
     /// `v0.8.9+commit.e5eed63a`
     /// `0.8.4+commit.dea1b9ec`
+    /// `0.8.4-beta.16+commit.dea1d`
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        let (_prefix, major, minor, patch, commit_hash) = sscanf::scanf!(
+        let (_prefix, major, minor, patch, pre, commit) = sscanf::scanf!(
             s,
-            "{:/v?/}{}.{}.{}+commit.{}",
+            "{:/v?/}{}.{}.{}{:/.*/}+commit.{:/[A-Fa-f0-9]+/}",
```

Comment:
Why not
```suggestion
            "{:/v?/}{}.{}.{}-{:/.*/}+commit.{:/[A-Fa-f0-9]+/}",
```
?

Reply:
because minus sign is optional

Reply:
it could be 
`{:/v?/}{}.{}.{}{:/-?/}{:/.*/}`, but I decided do it this way

Reply:
It's a small detail, but, according to the specification, Prerelease should be prefixed by `-` (https://docs.rs/semver/latest/semver/struct.Prerelease.html). Current regex allow to specify prerelease part without that prefix. Your second suggestion also does not fix that case.

Reply:
`{:/v?/}{}.{}.{}{:/(-.*)?/}`